library/alloc/src/collections/binary_heap.rs

```diff
@@ -889,6 +889,29 @@ impl<T> BinaryHeap<T> {
         self.data.shrink_to(min_capacity)
     }
 
+    /// Returns a slice of all values in the underlying vector, in arbitrary
+    /// order.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(binary_heap_as_slice)]
+    /// use std::collections::BinaryHeap;
+    /// let heap = BinaryHeap::from(vec![1, 2, 3, 4, 5, 6, 7]);
+    /// let slice = heap.as_slice();
+    ///
+    /// // Will print in some order
+    /// for x in slice {
+    ///     println!("{}", x);
+    /// }
```

Comment:
This is going to need a more realistic example. The point of example code is chiefly to show **why you would want** to call the method, not **how** to perform a method call. If someone is reading this documentation, we can assume they know how to make method calls. The example here does nothing to illustrate **why** `as_slice` would be used, since if you needed to do this, you should be writing `for x in &heap {...}` instead.

Reply:
@dtolnay Good question. I had bumped into it when I needed to [efficiently serialize BinaryHeap](https://github.com/near/borsh-rs/blob/c62cdfbd10d4a17fc877809eba4ccb65e866d5f8/borsh/src/ser/mod.rs#L248-L259). Some APIs (`writer::write_all` in my case) may expect a slice of data, and it is going to be quite inefficient to clone the data or iterate over it item by item (when you only have a reference to the BinaryHeap and cannot use `to_vec`)